### PR TITLE
Disable instrumentation according with SDK config

### LINKED ISSUE #43502: OpenTelemetry instrumentation still active after `quarkus.otel.sdk.disabled=true`
### Describe the bug

OpenTelemetry instrumentation is still active even when the SDK is disabled, therefore performing a non trivial amount of work and allocations that are not needed.

See the following flamegraph:
Screenshot 2024-09-25 at 17 46 14

This must be corrected.

### Expected behavior

When `quarkus.otel.sdk.disabled=true` the amount of OpenTelemetry related work must be near zero.

### Actual behavior

14,23% of all samples in the above flamegraph are OpenTelemetry related.

### How to Reproduce?

Using this project: https://github.com/brunobat/quarkus-observability-perf
Set `quarkus.otel.sdk.disabled=true`
Executing this hyperfoill test script to generate loa: https://github.com/brunobat/quarkus-observability-perf/blob/main/hyperfoil-scripts/quarkus-otel-perf.hf.yaml
And profiling the the execution with async profiler.

Data of the test execution can be seen performance test environment. Please contact me for details.

This affects all versions of the OpenTelemetry extension.

### PR DESCRIPTION
**Pull request by brunobat**
Fixes #43502

### CODE REVIEW

**Comment by brunobat** [comment]
This could be a candidate to backport, but not in the next minor because I want to have it in the wild for a bit, @gsmet.

File: extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryDisabledSdkTest.java [hunk 1]
```diff
@@ -8,8 +10,13 @@
...
 import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.quarkus.opentelemetry.runtime.config.runtime.OTelRuntimeConfig;
 import io.quarkus.opentelemetry.runtime.exporter.otlp.tracing.LateBoundBatchSpanProcessor;
+import io.quarkus.opentelemetry.runtime.tracing.intrumentation.vertx.HttpInstrumenterVertxTracer;
 import io.quarkus.test.QuarkusUnitTest;
+import io.vertx.core.spi.observability.HttpRequest;
+import io.vertx.core.spi.observability.HttpResponse;
 
 @Disabled("Not implemented")
```
Comment (radcortez) [inline extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryDisabledSdkTest.java:21]: Disabled, but the PR contains changes. Is this ok?

Comment (brunobat) [reply extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryDisabledSdkTest.java:21]: Nice catch. Must be removed.

**Review by radcortez — APPROVED** [review]

**Comment by brunobat** [comment]
@gsmet added this one for backport

**Pull request merged**